.github/workflows/redfield-sanity-check.yml

```diff
@@ -0,0 +1,39 @@
+# Syntax: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+# .NET CLI: https://docs.microsoft.com/dotnet/core/tools/
+# Description: The purpose of this build is to build and test with redfield flag.
+
+name: Redfield Sanity Check
+
+on:
+  push:
+    branches: [ master, develop ]
+  pull_request:
+    branches: [ master, develop ]
+
+jobs:
+  build-test:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        framework: [netcoreapp3.1,net5.0]
+        include: 
+            - os: ubuntu-latest
+              args: "--filter TestCategory!=WindowsOnly"
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: restore (Base sln)
+      run: dotnet restore ./BASE/Microsoft.ApplicationInsights.sln
+
+    - name: Build
+      run: dotnet build -p:Redfield=True ./BASE/Microsoft.ApplicationInsights.sln --configuration Release --no-restore
```

Comment:
only building Base sdk?

Reply:
Running our full test suite here will fail.
We have too many flaky tests and Github Actions does not have a mechanism for retries. 
(I'm investigating this in #2345)
If we want to run more test projects here, we must filter those tests.